DepotDownloader/ContentDownloader.cs

```diff
@@ -1004,6 +1005,34 @@ private static async Task DownloadSteam3Async( uint appId, List<DepotDownloadInf
 
                 await Task.WhenAll( tasks ).ConfigureAwait( false );
 
+                // Check for deleted files if updating the depot.
+                if ( oldProtoManifest != null )
+                {
+                    var oldfilesAfterExclusions = oldProtoManifest.Files.AsParallel().Where( f => TestIsFileIncluded( f.FileName ) ).ToList();
+
+                    foreach ( var file in oldfilesAfterExclusions )
+                    {
+                        // Delete it if it's in the old manifest AND not in the new manifest AND not in any of the previous depots.
+                        var newManifestFile = filesAfterExclusions.SingleOrDefault( f => f.FileName == file.FileName );
+                        if ( newManifestFile == null )
+                            continue;
+
+                        var previousFile = previousFiles.SingleOrDefault( f => f.FileName == file.FileName );

```

Comment:
This is incorrect to check here, as `previousFiles` is being filled within the same loop.

Reply:
It's being filled with external loop after the current depot is handled so this snippet checks files from previous depots. I'm pretty sure this is correct.